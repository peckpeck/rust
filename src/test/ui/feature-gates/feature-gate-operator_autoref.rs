#[derive(Debug)]
struct X {
}

impl std::ops::Add<&X> for &X {
    type Output = X;
    fn add(self, _rhs: &X) -> X {
        X { }
    }
}

fn main() {
    let a = X { };
    let b = X { };
    println!("x: {:?}", a+b); //~ ERROR cannot add `X` to `X`
}
