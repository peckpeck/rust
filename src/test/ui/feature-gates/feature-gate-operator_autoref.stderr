error[E0369]: cannot add `X` to `X`
  --> $DIR/feature-gate-operator_autoref.rs:15:26
   |
LL |     println!("x: {:?}", a+b);
   |                         -^- X
   |                         |
   |                         X
   |
note: an implementation of `Add<_>` might be missing for `X`
  --> $DIR/feature-gate-operator_autoref.rs:2:1
   |
LL | struct X {
   | ^^^^^^^^ must implement `Add<_>`
note: the following trait must be implemented
  --> $SRC_DIR/core/src/ops/arith.rs:LL:COL
   |
LL | / pub trait Add<Rhs = Self> {
LL | |     /// The resulting type after applying the `+` operator.
LL | |     #[stable(feature = "rust1", since = "1.0.0")]
LL | |     type Output;
...  |
LL | |     fn add(self, rhs: Rhs) -> Self::Output;
LL | | }
   | |_^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0369`.
